front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect, useMemo } from 'react';
+
+import type { ActionCreatorWithPayload } from '@reduxjs/toolkit';
+import { isNil } from 'lodash';
+import { useTranslation } from 'react-i18next';
+import nextId from 'react-id-generator';
+
+import type { SearchResultItemOperationalPoint } from 'common/api/osrdEditoastApi';
+import SelectSNCF from 'common/BootstrapSNCF/SelectSNCF';
+import useSearchOperationalPoint, {
+  MAIN_OP_CH_CODES,
+} from 'common/Map/Search/useSearchOperationalPoint';
+import type { PathStep } from 'reducers/osrdconf/types';
+import { useAppDispatch } from 'store';
+
+import StdcmSuggestions from './StdcmSuggestions';
+
+type UpdatePointActions =
+  | ActionCreatorWithPayload<PathStep | null, 'stdcmConf/updateOriginV2'>
+  | ActionCreatorWithPayload<PathStep | null, 'stdcmConf/updateDestinationV2'>;
+
+type StdcmOperationalPointProps = {
+  updatePoint: UpdatePointActions;
+  point: PathStep | null;
+  isPending?: boolean;
+};
+
+function formatChCode(chCode: string) {
+  return MAIN_OP_CH_CODES.includes(chCode) ? 'BV' : chCode;
+}
+
+const StdcmOperationalPoint = ({ updatePoint, point, isPending }: StdcmOperationalPointProps) => {
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation('stdcm');
+
+  const {
+    searchTerm,
+    chCodeFilter,
+    sortedSearchResults,
+    filteredAndSortedSearchResults,
+    setSearchTerm,
+    setChCodeFilter,
+  } = useSearchOperationalPoint({ initialSearchTerm: point?.name, initialChCodeFilter: point?.ch });
+
+  const operationalPointsSuggestions = [...sortedSearchResults].reduce(
+    (acc, p) => {
+      const newObject = {
+        label: [p.trigram, p.name].join(' '),
+        value: p.name,
+        uic: p.uic,
+      };
+      const isDuplicate = acc.some((pr) => pr.label === newObject.label);
+      if (!isDuplicate) acc.push(newObject);
+      return acc;
+    },
+    [] as { label: string; value: string; uic: number }[]
+  );
+
+  const sortedChOptions = useMemo(
+    () =>
+      [...sortedSearchResults].reduce(
```

Comment:
reduce does not mutate the array, so the copy of the dict is not necessary
```suggestion
      sortedSearchResults.reduce(
```